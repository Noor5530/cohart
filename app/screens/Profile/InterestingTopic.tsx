import Circle from 'assets/Circle.png';
import Community from 'assets/Community.png';
import Inspiration from 'assets/Inspiration.png';
import StarIconFull from 'assets/StarIconFull.png';

import { appFonts } from 'components/text';
import { placeHolderColor, primary, textColor } from 'config/colors';
import React, { useEffect, useRef, useState } from 'react';
import {
    Animated,
    FlatList,
    Pressable,
    StatusBar,
    StyleSheet,
} from 'react-native';
import { Button } from 'react-native-paper';
import { useDispatch, useSelector } from 'react-redux';
import { saveTopicRequest } from 'store/actions/userActions';

import { iconColor } from '../../config/colors';
import AppState from '../../models/reducers';
import ProfileHeader from './components/ProfileHeader';

const data = [
    {
        id: 1,
        name: 'Building My Community',
        tag: Community,
    },
    {
        id: 2,
        name: 'Showcasing My Work',
    },
    {
        id: 3,
        name: 'Buying Artwork',
    },
    {
        id: 4,
        name: 'Inspiration',
        tag: Inspiration,
    },
    {
        id: 5,
        name: 'Sharing Ideas & Concepts',
        tag: Community,
    },
    {
        id: 6,
        name: 'Meeting  Artists',
        tag: Circle,
    },
    {
        id: 7,
        name: 'Meeting Collectors',
    },
    {
        id: 8,
        name: 'Collaboration',
        tag: Community,
    },
    {
        id: 9,
        name: 'Art News',
    },
    {
        id: 10,
        name: 'Workshops & Talks',
        tag: StarIconFull,
    },
    {
        id: 11,
        name: 'Reselling Artwork',
    },
    {
        id: 12,
        name: 'Exploration',
    },
    {
        id: 13,
        name: 'I Don’t Know Yet',
    },
];
const Profile: React.FC = () => {
    const interests = useSelector((state: AppState) => state.user.interests);
    const [selectedTopic, setSelectedTopic] = useState(interests);

    const onSave = () => {
        dispatch(saveTopicRequest(selectedTopic));
    };
    const dispatch = useDispatch();
    const renderTopics = ({ item, index }: { item: any, index: number }) => {
        return (
            <Tags
                item={item}
                selectedTopic={selectedTopic}
                setSelectedTopic={setSelectedTopic}
                index={index}
            />
        );
    };
    return (
        <>
            <StatusBar backgroundColor={textColor} barStyle="dark-content" />

            <ProfileHeader
                description={'[ SELECT 3 OR MORE ]'}
                heading={'I’M INTERESTED IN..'}
            />

            <FlatList
                contentContainerStyle={{
                    paddingHorizontal: 38,
                    paddingTop: 28,
                    paddingBottom: 100,
                }}
                data={data}
                renderItem={renderTopics}
                keyExtractor={(item, index) => `id_${index}`}
            />
            {selectedTopic.length > 1 ? (
                <Button
                    style={styles.saveButton}
                    onPress={onSave}
                    mode="text"
                    labelStyle={styles.buttonText}>
                    START DISCOVERING
                </Button>
            ) : null}
        </>
    );
};

interface ITagsProps {
  setSelectedTopic: (topic: any) => void;
  item: any; 
  selectedTopic: any;
  index: number;
}

const Tags = (props: ITagsProps) => {
  const { item, selectedTopic, index } = props;
  // const color = useSharedValue(colors[0]);
  const animation = useRef(new Animated.Value(0));

    const interests = useSelector((state: AppState) => state.user.interests);

    const handleAnimation = () => {
        Animated.timing(animation.current, {
            toValue: 1,
            duration: 800,
            useNativeDriver: false,
        }).start();
    };
    const fadeOut = () => {
        Animated.timing(animation.current, {
            toValue: 0,
            duration: 800,
            useNativeDriver: false,
        }).start();
    };
    const fadeText = animation.current.interpolate({
        inputRange: [0, 1],
        outputRange: [placeHolderColor, '#000000'],
    });
    useEffect(() => {
        interests.map(tag => {
            if (tag == item.name) {
                handleAnimation();
            }
        });
    }, []);
    const toggleSelection = () => {
        let index = -1;
        index = selectedTopic.findIndex((tag: string) => {
            return tag == item.name;
        });
        if (index !== -1) {
            const data = selectedTopic.filter((tag: string) => {
                return tag !== item.name;
            });

            props.setSelectedTopic(data);
            fadeOut();
        } else {
            const temp = [...props.selectedTopic, item.name];
            props.setSelectedTopic(temp);
            handleAnimation();
        }
    };
    return (
        <Pressable style={styles.row} onPress={toggleSelection}>
            <Animated.Text
                style={[styles.animatedText, { color: fadeText }]}
                key={`id_${index}`}>
                {item.name}
            </Animated.Text>
            {item.tag && (
                <Animated.Image
                    resizeMode="contain"
                    style={[
                        styles.animatedImage,
                        { opacity: animation.current },
                    ]}
                    source={item.tag}
                />
            )}
        </Pressable>
    );
};

export default Profile;
const styles = StyleSheet.create({
    animatedImage: {
        position: 'absolute',
        top: 10,
        left: 210,
        width: 55,
        height: 50,
    },
    animatedText: {
        width: 298,
        fontFamily: appFonts.InterRegular,
        fontSize: 40,
        fontWeight: 'bold',
        paddingBottom: 25,
    },
    row: {
        flexDirection: 'row',
    },
    saveButton: {
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: primary,
        borderRadius: 39,
        alignSelf: 'center',
        position: 'absolute',
        bottom: 50,
    },
    buttonText: {
        color: iconColor,
        fontSize: 13,
        paddingVertical: 7,
        fontWeight: '700',
        textAlign: 'center',
        fontFamily: appFonts.InterRegular,

        paddingHorizontal: 20,
    },
});
