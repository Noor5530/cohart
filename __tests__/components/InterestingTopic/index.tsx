import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import InterestingTopic from "../../../app/screens/FirstLogin/components/InterestingTopic";
jest.useFakeTimers();

jest.mock('react-redux', () => ({
    useDispatch: () => {},
    useSelector: () => ({
      user: {
        interests: []
    },
    }),
  }));

  jest.mock("react-native-paper/src/components/Snackbar", () => "Snackbar");
  jest.mock("react-native-paper/src/components/Button", () => "Button");
test("renders correctly", () => {
  const tree = renderer.create(<InterestingTopic />).toJSON();

  expect(tree).toMatchSnapshot();
});


