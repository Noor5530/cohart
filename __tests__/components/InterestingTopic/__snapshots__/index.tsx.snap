// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#E6FF00",
          "height": undefined,
          "justifyContent": "flex-end",
          "paddingBottom": 30,
        },
        Object {
          "width": 750,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/profileHeaderIcon.png",
        }
      }
      style={
        Object {
          "height": undefined,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": undefined,
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Inter-Bold",
            "fontSize": undefined,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "height": 19,
            "lineHeight": 19,
            "paddingLeft": 38,
            "textTransform": "uppercase",
          }
        }
      >
        I’M INTERESTED IN..
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": undefined,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "paddingLeft": 38,
            "textTransform": "uppercase",
          }
        }
      >
        [ SELECT 3 OR MORE ]
      </Text>
    </View>
  </View>,
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 100,
        "paddingHorizontal": 38,
        "paddingTop": 28,
      }
    }
    data={
      Array [
        Object {
          "id": 1,
          "name": "Building My Community",
          "tag": Object {
            "testUri": "../../../app/assets/Community.png",
          },
        },
        Object {
          "id": 2,
          "name": "Showcasing My Work",
        },
        Object {
          "id": 3,
          "name": "Buying Artwork",
        },
        Object {
          "id": 4,
          "name": "Inspiration",
          "tag": Object {
            "testUri": "../../../app/assets/Inspiration.png",
          },
        },
        Object {
          "id": 5,
          "name": "Sharing Ideas & Concepts",
          "tag": Object {
            "testUri": "../../../app/assets/Community.png",
          },
        },
        Object {
          "id": 6,
          "name": "Meeting  Artists",
          "tag": Object {
            "testUri": "../../../app/assets/Circle.png",
          },
        },
        Object {
          "id": 7,
          "name": "Meeting Collectors",
        },
        Object {
          "id": 8,
          "name": "Collaboration",
          "tag": Object {
            "testUri": "../../../app/assets/Community.png",
          },
        },
        Object {
          "id": 9,
          "name": "Art News",
        },
        Object {
          "id": 10,
          "name": "Workshops & Talks",
          "tag": Object {
            "testUri": "../../../app/assets/StarIconFull.png",
          },
        },
        Object {
          "id": 11,
          "name": "Reselling Artwork",
        },
        Object {
          "id": 12,
          "name": "Exploration",
        },
        Object {
          "id": 13,
          "name": "I Don’t Know Yet",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Building My Community
          </Text>
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/Community.png",
              }
            }
            style={
              Object {
                "height": 50,
                "left": 210,
                "opacity": 0,
                "position": "absolute",
                "top": 10,
                "width": 55,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Showcasing My Work
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Buying Artwork
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Inspiration
          </Text>
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/Inspiration.png",
              }
            }
            style={
              Object {
                "height": 50,
                "left": 210,
                "opacity": 0,
                "position": "absolute",
                "top": 10,
                "width": 55,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Sharing Ideas & Concepts
          </Text>
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/Community.png",
              }
            }
            style={
              Object {
                "height": 50,
                "left": 210,
                "opacity": 0,
                "position": "absolute",
                "top": 10,
                "width": 55,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Meeting  Artists
          </Text>
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/Circle.png",
              }
            }
            style={
              Object {
                "height": 50,
                "left": 210,
                "opacity": 0,
                "position": "absolute",
                "top": 10,
                "width": 55,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Meeting Collectors
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Collaboration
          </Text>
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/Community.png",
              }
            }
            style={
              Object {
                "height": 50,
                "left": 210,
                "opacity": 0,
                "position": "absolute",
                "top": 10,
                "width": 55,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Art News
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.2)",
                "fontFamily": "Inter-Regular",
                "fontSize": 40,
                "fontWeight": "bold",
                "paddingBottom": 25,
                "width": 298,
              }
            }
          >
            Workshops & Talks
          </Text>
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/StarIconFull.png",
              }
            }
            style={
              Object {
                "height": 50,
                "left": 210,
                "opacity": 0,
                "position": "absolute",
                "top": 10,
                "width": 55,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>,
]
`;
